### PR TITLE
10-alex-module-5

### LINKED ISSUE #10: Alex - Module 5
Changes to be applied :

- Check inheritance vs composition.
- Singleton pattern.
- Single responsibility principle: A class should have one reason to change, or one responsibility.
- Open-closed principle: Objects should be open for extension but closed for modification
- Liskov substitution principle: A principle of OOP. 
- Interface segregation principle: A principle of OOP.
- Dependency inversion principle: A principle of OOP.

### PR DESCRIPTION
**Pull request by Caticatt**

### CODE REVIEW

**Comment by BorjaDelgadoDev** [comment]
@Caticatt your branch 10-alex-module-5 should go against develop_alex

File: build.gradle.kts [hunk 1]
```diff
@@ -3,4 +3,20 @@ plugins {
...
+
+subprojects {
+    apply(plugin = "io.gitlab.arturbosch.detekt")
+
+    detekt {
+        config.setFrom(file("$rootDir/config/detekt/detekt.yml"))
+
+        dependencies {
+            //noinspection UseTomlInstead
+            detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.23.1")
```
Comment (diego-matesanz) [inline build.gradle.kts:18]: I had the same problem here, using libs version catalog throws an execption at building time. In my case it has been solved using the toml version catalog via a variable like this:

```
val detektFormatting = libs.detekt.formatting

subprojects {
    apply(plugin = "io.gitlab.arturbosch.detekt")
    detekt {
        config.setFrom(files("$rootDir/config/detekt/detekt.yml"))

        dependencies {
            detektPlugins(detektFormatting)
        }
    }
}
```

**Review by diego-matesanz — APPROVED** [review]
Nice job! Just the comment about version catalogs. Hope it works for you!

**Review by cristiangoncas — APPROVED** [review]
Looks good!

**Review by BorjaDelgadoDev — APPROVED** [review]
Approved!

**Pull request closed without merging**